owncloudApp/src/main/java/com/owncloud/android/ui/activity/FolderPickerActivity.java

```diff
@@ -509,4 +532,19 @@ private void refreshList(boolean ignoreETag) {
             }
         }
     }
+
+    public enum PickerMode {
+        MOVE, COPY, CAMERA_FOLDER;
+
+        public Integer getButtonString(PickerMode pickerMode) {
+            switch (pickerMode) {
```

Comment:
This is a class function. You don't need to receive the picker mode from parameters, because you already know which picker mode is when u call it. 

Try next code and let me know what happens 👍  (This will simplify setActionButtonText code too)
```suggestion
        public Integer getButtonString() {
            switch (this) {
```

Reply:
OMG 😱  Marvelous!! Thanks.